mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -744,7 +744,13 @@ private[classification] class MultiClassSummarizer extends Serializable {
   def countInvalid: Long = totalInvalidCnt
 
   /** @return The number of distinct labels in the input dataset. */
-  def numClasses: Int = distinctMap.keySet.max + 1
+  def numClasses: Int = try {
+    distinctMap.keySet.max + 1
+  } catch {
+    case e: UnsupportedOperationException =>
```

Comment:
@srowen Let me do some test about return 0 directly when the hashtable is empty and make changes accordingly. Thanks for your review!